### PR TITLE
feat: properly handle closing viewers for grid layout

### PR DESCRIPTION
**Pull request by iisakkirotko**

### CODE REVIEW

**Comment by iisakkirotko** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/widgetti/glue-solara/11?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11** <a href="https://app.graphite.dev/github/pr/widgetti/glue-solara/11?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10** <a href="https://app.graphite.dev/github/pr/widgetti/glue-solara/10?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9** <a href="https://app.graphite.dev/github/pr/widgetti/glue-solara/9?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iisakkirotko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: glue_solara/app.py [hunk 1]
```diff
@@ -305,6 +308,15 @@ def on_viewer_close(index: int):
         new_mdi_layouts.pop(index)
         mdi_layouts.set(new_mdi_layouts)
 
+        # Remove the viewer from grid layouts
+        new_grid_layout = grid_layout.value
```
Comment (maartenbreddels) [inline glue_solara/app.py:312]:
```suggestion
        new_grid_layout = copy.deepcopy(grid_layout.value)
```

**Review by maartenbreddels — COMMENTED** [review]
looks good, except the mutation of state.

File: glue_solara/app.py (hunk 1 above)
Comment (iisakkirotko) [reply glue_solara/app.py]: I guess we should also change it [here](https://github.com/widgetti/glue-solara/blob/1f3faf39bc10bf00360f2a329ab911711dff2e10/glue_solara/app.py#L304)?

**Comment by iisakkirotko** [comment]
### Merge activity

* **Oct 25, 7:34 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/widgetti/glue-solara/11).

**Pull request merged**